airbyte-cdk/python/unit_tests/test_connector.py

```diff
@@ -71,3 +80,53 @@ def test_write_config(integration, mock_config):
     integration.write_config(mock_config, str(config_path))
     with open(config_path, "r") as actual:
         assert mock_config == json.loads(actual.read())
+
+
+class TestConnectorSpec:
+    CONNECTION_SPECIFICATION = {
+        "type": "object",
+        "required": ["api_token"],
+        "additionalProperties": False,
+        "properties": {"api_token": {"type": "string"}},
+    }
+
+    @pytest.fixture
+    def use_json_spec(self):
+        spec = {
+            "documentationUrl": "https://airbyte.com/#json",
+            "connectionSpecification": self.CONNECTION_SPECIFICATION,
+        }
+
+        json_path = os.path.join(SPEC_ROOT, "spec.json")
+        with open(json_path, "w") as f:
+            f.write(json.dumps(spec))
+        yield
+        os.remove(json_path)
```

Comment:
probably preferable to use `tempfile` here. We're already using it above in `TestAirbyteSpec.test_from_file`. Adding your new unit tests into that class you could combine and tidy everything up.